include/soci/odbc/soci-odbc.h

```diff
@@ -84,6 +84,12 @@ class odbc_standard_type_backend_base
         max_bigint_length = 21
     };
 
+    // IBM DB2 driver is not compliant to ODBC spec for indicators in 64bit
+    // SQLLEN is still defined 32bit (int) but spec requires 64bit (long)
+    inline bool requires_not_spec_compliant_indicators() const;
+    inline SQLLEN get_spec_compliant_sqllen_from_value(const SQLLEN val) const;
```

Comment:
To me, those may not be good names.
The functions cope with DB2-specific behaviour but the names are generic and, despite the comments, may suggest it's not just DB2.
OTOH, there may be other DBMS which break the ODBC spec in the same way.

What about renaming indicate it is about 32-bit `SQLLEN` specifically, to something like:

```
inline bool requires_noncompliant_32bit_sqllen() const;
inline bool get_sqllen_from_noncompliant_32bit_sqllen() const;
```

The key `spec` is not necessary - the backend scope and `SQLLEN` should be enough to indicate it is about ODBC compliance.

Other than that suggestin, it LGTM. Thanks!

I will leave the final decision about mergint to @vadz - he's been maintaining the backend mostly.

Reply:
Good point to rename the methods, I can do this. Do you prefer it for the vector methods too?

Reply:
Sorry, I missed the naming for the vector variants.

I would refactor it this way:

* The predicate is given name that indicates what problem it solves:

```
requires_noncompliant_32bit_sqllen
```

* The getters/setters with generic and short names:

```
get_sqllen_from_value
get_sqllen_from_vector_at
set_sqllen_into_vector_at
```

For symmetry, there could also be `set_sqllen_into_value`, if needed.

IMHO, the getters/setters do not need to indicate why/how they are sanitizing - they equally could be used to have some run-time checks/assertions of buffers with indicators, etc.
We simply wrap the non-trivial get/set logic into functions.